### PR TITLE
Add a lock to bottom sheet to prevent repeated tap bugs

### PR DESCRIPTION
**Pull request by atabel**

### CODE REVIEW

File: src/bottom-sheet.ts [hunk 1]
```diff
@@ -28,8 +28,33 @@ type SheetUI = Readonly<{
...
+    payload: SheetUI,
+): Promise<SheetResponse | SheetLockedResponse> => {
+    if (sheetLock) {
+        return Promise.resolve({action: 'LOCKED', result: []});
+    }
+
+    sheetLock = true;
+    const tid = setTimeout(() => {
+        sheetLock = false;
+    }, 1000);
```
Comment (wyunreal) [inline src/bottom-sheet.ts:45]: maybe 3 seconds or so is better than just a second ?

**Review by wyunreal — APPROVED** [review]

File: src/bottom-sheet.ts (hunk 1 above)
Comment (atabel) [reply src/bottom-sheet.ts:45]: the idea is to have a lock for the time between the user taps and the bottom sheet appears. I think that the bottom sheet should appear in less than 1 second

File: src/bottom-sheet.ts [hunk 2]
```diff
@@ -28,8 +28,33 @@ type SheetUI = Readonly<{
...
+let sheetLock = false;
+
+type SheetLockedResponse = {action: 'LOCKED'; result: Array<any>};
+
+export const bottomSheet = async (
+    payload: SheetUI,
+): Promise<SheetResponse | SheetLockedResponse> => {
+    if (sheetLock) {
+        return Promise.resolve({action: 'LOCKED', result: []});
+    }
```
Comment (pladaria) [inline src/bottom-sheet.ts:40]: Wouldn't be better to reject this?

Comment (atabel) [reply src/bottom-sheet.ts:40]: CPS code is catching any error and showing the old picker in that case

**Review by pladaria — APPROVED** [review]

File: src/bottom-sheet.ts (hunk 2 above)
Comment (atabel) [reply src/bottom-sheet.ts]: changed to an error

**Pull request merged**